tests/capture/test_capture_mid_measure.py

```diff
@@ -339,24 +336,25 @@ def f(x):
             qml.capture.disable()
             expected = f(phi)
 
-            if mp_fn is qml.expval:
-                assert qml.math.allclose(res, expected, atol=1 / qml.math.sqrt(shots), rtol=0.1)
-            elif mp_fn is qml.var:
-                assert qml.math.allclose(res, expected, atol=1 / qml.math.sqrt(shots), rtol=0.1)
-            elif mp_fn is qml.probs:
-                assert qml.math.allclose(res, expected, atol=1 / qml.math.sqrt(shots), rtol=0.1)
-            else:
-                # mp_fn is qml.sample
+            if mp_fn is qml.sample:
                 ones_count = jnp.sum(res == 1)
                 minus_ones_count = jnp.sum(res == -1)
-                theoretical_ones_count = shots * (qml.math.cos(phi / 2) ** 2)
-                theoretical_minus_ones_count = shots * (qml.math.sin(phi / 2) ** 2)
                 if reset:
                     # either all ones or all minus ones
                     assert ones_count == 0 or minus_ones_count == 0
                 else:
-                    assert qml.math.allclose(ones_count, theoretical_ones_count, atol=5)
-                    assert qml.math.allclose(minus_ones_count, theoretical_minus_ones_count, atol=5)
+                    sample_avg = 1.0 * (ones_count - minus_ones_count) / shots
+                    # Check the sample average instead of individual counts
+                    # For eigenvalues {+1, -1}, the sample average should be p_plus - p_minus
+                    # with std = 2*sqrt(p_plus*p_minus/shots)
+                    p_plus = qml.math.cos(phi / 2) ** 2
+                    p_minus = 1 - p_plus
+                    expected_avg = p_plus - p_minus
+                    std_sample_avg = 2 * qml.math.sqrt(p_plus * p_minus / shots)
+                    atol = 3 * std_sample_avg
+                    assert qml.math.allclose(sample_avg, expected_avg, atol=atol, rtol=0)
```

Comment:
Optional, a bit easier to immediately understand
```suggestion
                    sample_expected_avg = 1.0 (ones_count / shots) + (-1.0) * (minus_ones_count / shots)
                    # Check the sample average instead of individual counts
                    # For eigenvalues {+1, -1}, the sample average should be p_plus - p_minus
                    # with std = 2*sqrt(p_plus*p_minus/shots)
                    p_plus = qml.math.cos(phi / 2) ** 2
                    p_minus = 1 - p_plus
                    expected_avg = p_plus - p_minus
                    std_sample_avg = 2 * qml.math.sqrt(p_plus * p_minus / shots)
                    atol = 3 * std_sample_avg
                    assert qml.math.allclose(sample_expected_avg, expected_avg, atol=atol, rtol=0)
```